src/main/java/io/cdap/plugin/aws/s3/source/S3BatchSource.java

```diff
@@ -202,6 +209,44 @@ public void validate(FailureCollector collector) {
       }
     }
 
+    private void validatePath(FailureCollector collector) {
+      if (!containsMacro("path") && (!path.startsWith("s3a://") && !path.startsWith("s3n://"))) {
+        collector.addFailure("Path must start with s3a:// or s3n://.", null).withConfigProperty(NAME_PATH);
+      }
+      if (!containsMacro("path") && path.startsWith("s3n://") && !Strings.isNullOrEmpty(connection.getSessionToken())) {
+        collector.addFailure("Temporary credentials are only supported for s3a:// paths.", null)
+          .withConfigProperty(NAME_PATH);
+      }
+    }
+
+    private void validateCredentials(FailureCollector collector) {
+
+      AmazonS3 s3 = getS3Client(Regions.DEFAULT_REGION.getName());
+      try {
+        s3.listBuckets();
```

Comment:
What if user does not have permission to list buckets and only have permission on a particular bucket?

Was this permission always needed before in S3 batch source?

Reply:
@albertshau wdyt?

Reply:
Generally we make a call to `detectSchema()` in GCS and other sources which should be enough to validate credentials, aren't we doing similar in S3?

Reply:
> What if user does not have permission to list buckets and only have permission on a particular bucket?

Yes, the user may not have the permission for this operation. We're going ahead with the assumption that the user should already have access to list the buckets.  As per the offline discussion with Saurabh, I've replaced the `listBuckets()` API with the `getCallerIdentity()` API from AWS STS SDK. dbaf2f1183f1c59b081da9170b80d6ef24acc0f7

No permission is required for this operation.
https://docs.aws.amazon.com/STS/latest/APIReference/API_GetCallerIdentity.html

> Was this permission always needed before in S3 batch source?

As of now, the user is required to have this permission to view the buckets through Wrangler

Reply:
> Generally we make a call to `detectSchema()` in GCS and other sources which should be enough to validate credentials, aren't we doing similar in S3?

Here, the schema is fetched through the methods called in the superclass: `AbstractFileSource` 
To validate credentials, we need to create a client and call AWS API.